File: Practice/06/Phyton/666.py
import math

print("Введите 3 вещественных числа")

a = int(input())
b = int(input())
c = int(input())

if a == 0:
    if b == 0:
        print("Корней нет.")
    else:
        print("x =", -c / b)
else:
    d = b**2 - 4*a*c
    if d < 0:
        print("Корней нет.")
    elif d == 0:
        print("x =", -b/(2*a))
    else:
        d_sqrt = math.sqrt(d)
        x1 = (-b + d_sqrt) / (2 * a)
        x1 = (-b - d_sqrt) / (2 * a)
        print("x1 =", x1,"x2 =", x2)

File: Practice/06/C++/666.cpp

#include <iostream>
#include <math.h>

using namespace std;
int a, b, c, d;
float x1, x2;
int main() {
    std::cout << ("Введите коэфициент a") << "\n";
    std::cin >> a;
    std::cout << ("Введите коэфициент b") << "\n";
    std::cin >> b;
    std::cout << ("Введите коэфициент c") << "\n";
    std::cin >> c;
    d = (b * b) - (4 * (a * c));
    if (d < 0) {
        std::cout << ("Нет корней");
    }
    else if (d = 0) {
        x1 = ((-b) + (sqrt(d)) / (2 * a));
        std::cout << ("Один корень") << "\n";
        std::cout << (x1);
    }
    else {
        x1 = ((-b) + (sqrt(d)) / (2 * a));
        x2 = ((-b) - (sqrt(d)) / (2 * a));
        std::cout << (x1);
        std::cout << (x2);
    }

}